include/rwahs/tasks/build

```diff
@@ -7,11 +7,108 @@ set -e
 set -u
 
 function build_showHelp {
-    echo "TASK DESCRIPTION:"
-    echo "    Create a new release, which is a tag in the code repository."
+    local programName toolsDir
+
+    # Function arguments
+    programName="${1}"
+    toolsDir="${2}"
+    shift 2
+
+    echo
+    echo "Create a new release, which is a tag in the code repository."
+    echo
+    echo "USAGE:"
+    echo "    ${programName} build -c <component> [-t <target-commitish>] [-r <release>]"
+    echo
+    echo "WHERE:"
+    echo "    -t <target-commitish> gives the branch or commit that should be released; if omitted, the default branch"
+    echo "         according to the component configuration (not the default branch configured in github) will be used."
+    echo "    -c <component> gives the name of the component to process."
+    echo "         Valid values: $(ls -m "${toolsDir}/config/${programName}/components")"
+    echo "    -r <release> gives the optional release tag override; if omitted, a value will be generated and output."
     echo
 }
 
 function build_runTask {
-    echo "Running 'build' task..."
+    local programName toolsDir ghApiRoot component releaseName releaseDate targetCommitish apiJsonTemplate apiJson error
+    local repo repoDefaultTargetCommitish targetParent
+
+    # Function arguments
+    programName="${1}"
+    toolsDir="${2}"
+    shift 2
+
+    # Default values
+    ghApiRoot="https://api.github.com"
+    releaseDate=$(date "+%Y/%m/%d %H:%M:%S")
+    apiJsonTemplate='{"tag_name":"release_%s","target_commitish":"%s","name":"%s","body":"Release tagged at %s","draft":false,"prerelease":false}'
+
+    # Parse parameters
+    while getopts :c:r:t: parameter; do
+        case "${parameter}" in
+            c)
+                component="${OPTARG}"
+            ;;
+            r)
+                releaseName="${OPTARG}"
+            ;;
+            t)
+                targetCommitish="${OPTARG}"
+            ;;
+            \?)
+                echo "Error: Invalid option '-${OPTARG}' provided to the 'build' task." >&2
+                build_showHelp "${programName}" "${toolsDir}" >&2
+                exit 1
+            ;;
+            :)
```

Comment:
The `parameter` variable is set to `?` if the flag is unsupported, or `:` if the flag requires an argument but none was found. The `?` is escaped for bash, otherwise it would match any character (glob style). In either case, `OPTARG` is set to the flag itself.